subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/tasks/CreateDependencyInfoFile.java

```diff
@@ -0,0 +1,70 @@
+package org.shipkit.internal.gradle.java.tasks;
+
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.SelfResolvingDependency;
+import org.json.simple.JsonArray;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.notes.util.IOUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+public class CreateDependencyInfoFile {
+
+    private static final String DESCRIPTION = "This file was generated by Shipkit Gradle plugin. See http://shipkit.org\n\n";
+
+    public void createDependencyInfoFile(CreateDependencyInfoFileTask task) {
+        JsonObject result = createJsonObject();
+        result.put("description", DESCRIPTION);
+
+        JsonArray dependencies = createJsonArray();
+
+        for (ResolvedArtifact artifact : task.getConfiguration().getResolvedConfiguration().getResolvedArtifacts()) {
```

Comment:
In fact, we only need the 1st level dependencies. I think that the code currently gets you full transitive graph. pom file only contains 1st level dependencies (direct).

However, maybe it's not that bad to include full transitive graph. It might be an interesting piece of information to keep in the jar.